cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/write_cloudwatch_agent_json.py

```diff
@@ -177,14 +178,68 @@ def filter_output_fields(configs):
     return [{desired_key: config[desired_key] for desired_key in desired_keys} for config in configs]
 
 
-def create_config(log_configs):
+def create_metrics_collected(selected_configs):
+    """Create the "metrics_collected" section in metrics configuration for selected metrics."""
+    metrics_collected = {}
+    for metric_config in selected_configs:
+        metric_key = metric_config["metric_type"]
+        metrics_collected[metric_key] = {
+            "measurement": metric_config["measurement"],
+            "metrics_collection_interval": metric_config.get(
+                "metrics_collection_interval", DEFAULT_METRICS_COLLECTION_INTERVAL
+            ),
+        }
+        if "resources" in metric_config.keys():
+            metrics_collected[metric_key]["resources"] = metric_config["resources"]
+        if "append_dimensions" in metric_config.keys() and "ClusterName" in metric_config["append_dimensions"]:
```

Comment:
nit: `if "append_dimensions" in metric_config.keys()` can be simplified as if "append_dimensions" in metric_config